### DISCUSSION TITLE
How to generate fixed-segregating-sites mutations

### DISCUSSION THREAD

**Discussion by haotianzh** [tskit-dev/msprime#1822] [Q&A]
ms has already had a function to produce mutation data with fixed SNP sites (For example, 200 sites). Is there any similar function in mspirme?

**Comment by jeromekelleher** [comment]
We don't provide this functionality @haotianzh as it is quite subtle, and actually very rarely what people want in practise. See #68 for a discussion.

If you describe your application we might be able to suggest an alternative solution.

**Comment by molpopgen** [comment]
For future reference: a user could, I think, mimic what `ms` does rather easily, at least for infiinitely-many sites:

For each tree:

1. The number of mutations, S, is Poisson given total time * mutation rate.  (Caveat: mutation rate must be scaled correctly...0
2. The mutations are placed onto branches according to a multinomial distribution whose coefficients are the branch lengths divided by the total time on the tree.

`ms` does exactly this, but used a Gaussian for S if total time * mutation rate > 30 or so.
